test/UnitTest/Microsoft.AspNet.OData.Test.Shared/ScaleDecimalTest.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using Microsoft.AspNet.OData.Builder;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Test.Abstraction;
+using Microsoft.AspNet.OData.Test.Builder.TestModels;
+using Microsoft.OData.Edm;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.OData.Test
+{
+    public class ScaleDecimalTest
+    {
+        [Fact]
+        public async Task ScaleOfDecimalProperty_ShouldBeCalculated_BasedOnScale()
+        {
+            const string Uri = "http://localhost/odata/Orders";
+            HttpClient client = GetClient();
+
+            HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Get, Uri);
+            request.Headers.Accept.Add(MediaTypeWithQualityHeaderValue.Parse("application/json;odata.metadata=full"));
+
+            // Act
+            HttpResponseMessage response = await client.SendAsync(request);
+
+            // Assert
+            IEnumerable<Order> orders = await ReadEntitiesFromResponse<Order>(response);
```

Comment:
@komdil Let's assume decimal values in the raw response appear as 10:00. The moment the payload is deserialized into CLR objects, there's no guarantee the scale is respected or preserved as the `TotalPrice` property is materialized. Consequently, `order.TotalPrice.ToString(CultureInfo.InvariantCulture)` won't necessarily yield "10.00"

An alternative approach (hopefully foolproof) is to read the response stream into a string and base your assert on that, i.e.
```csharp
var response = await client.SendAsync(request);
var responseContent = await response.Content.ReadAsStringAsync();
Assert.Equal("EXPECTED", responseContent);
```

If your fix is good then the array part of the payload would be expected to be something like this: 
`[{"OrderId":1,"TotalPrice":10.00},{"OrderId":2,"TotalPrice":10.00},{"OrderId":3,"TotalPrice":10.00}]`

Kindly let me know if you're in a position to try this to fix the failing test(s)